www/templates/survey/enketo/add-note-button.html

```diff
@@ -0,0 +1,5 @@
+<div class="enketo-button">
+  <button class="button icon ion-plus button-primary" ng-click="openPopover($event, trip, 'NOTES')">
+    <b>{{displayLabel()}}</b>
```

Comment:
Does Angularjs not have a way to define a property, cache it and then automatically recompute when its dependencies change?
I miss this from other frameworks

Reply:
It does do this, which is the `$digest`/`$apply` mechanism.
So if you define a variable in `$scope`, the value is cached unless you call a `$scope.$apply`, which will basically recompute the cached values using some kind of dependency graph.

But the point I was making here is that you are not using a variable, you are using a function. So while rendering, angular will *always* call the function since the function may return different results at different times. You want to replace all functions in the HTML (at least to the extent possible) with variables.

e.g. https://github.com/e-mission/e-mission-phone/commit/30561b94d8691e37fc3e50b8488f5da82f6dd815

Reply:
6888ec5c8e4594cc250eea791a669fa4c59131b5 handles this